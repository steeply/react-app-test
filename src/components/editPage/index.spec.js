/*eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import ConnectedEditPage from './index';

describe('ConnectedEditPage container', () => {
  const initialState = {
    models: {
      list: [
        { id: 1, modelName: 'name', properties: [] }
      ],
      isLoading: false
    }
  };

  const nextProps = {
    match: {
      params: {
        id: '1'
      }
    }
  };

  const mockStore = configureStore([thunk]);
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<ConnectedEditPage {...nextProps} store={store} />);
  });

  it('+++ check Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('+++ render the connected(EditPage) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('+++ check Prop matches with initialState', () => {
    expect(wrapper.prop('modelsList')).toEqual(initialState.models.list);
  });

  it('+++ is render tag <h1>', () => {
    const component = wrapper.dive();
    expect(component.find('h1').text()).toEqual('Редактирование модели (id: 1)');
  });

  it('checks componentDidMount is called', () => {
    const spy = jest.spyOn(ConnectedEditPage.prototype, 'componentDidMount');
    shallow(<ConnectedEditPage store={store} />);
    expect(spy).toHaveBeenCalled();
    ConnectedEditPage.prototype.componentDidMount.mockRestore();
  });

});

/*eslint-enable */