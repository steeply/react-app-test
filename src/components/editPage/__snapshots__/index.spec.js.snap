// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedEditPage container +++ check Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(EditPage)
    match={
      Object {
        "params": Object {
          "id": "1",
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addModel": [Function],
      "deleteModel": [Function],
      "editModel": [Function],
      "homePage": [Function],
      "isLoading": false,
      "loadModels": [Function],
      "match": Object {
        "params": Object {
          "id": "1",
        },
      },
      "modelsList": Array [
        Object {
          "id": 1,
          "modelName": "name",
          "properties": Array [],
        },
      ],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addModel": [Function],
        "deleteModel": [Function],
        "editModel": [Function],
        "homePage": [Function],
        "isLoading": false,
        "loadModels": [Function],
        "match": Object {
          "params": Object {
            "id": "1",
          },
        },
        "modelsList": Array [
          Object {
            "id": 1,
            "modelName": "name",
            "properties": Array [],
          },
        ],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
